### PR TITLE
Ascent

### PR DESCRIPTION
**Pull request by mclarsen**

### CODE REVIEW

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.H [hunk 1]
```diff
@@ -83,6 +83,17 @@ namespace amrex
                                 const Vector<IntVect> &ref_ratio,
                                 conduit::Node &bp_mesh);
 
+    // Creates an the parent-child relationships for a fab
+    //
+    void Nestsets(const int level,
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.H:88]: do we need this in the header, or can we keep only internal to the cpp file?

(I don't think we expect folks to call it themselves?)

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp [hunk 2]
```diff
@@ -14,6 +14,103 @@ using namespace conduit;
 
 namespace amrex {
 
+void paint_2d_nestsets(conduit::Node &domain,
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:17]: do we need the paint code here, or was it for debugging?

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp [hunk 3]
```diff
@@ -46,12 +143,14 @@ void FabToBlueprintTopology(const Geometry& geom,
...
+    // geom.CellSize()[k] == (level_z_max - level_z_min) / float64(level_nz)
     float64 level_dz = dims > 2 ? geom.CellSize()[2] : 0.0;
-    
+
     // now extract the FAB details
     const amrex::Box &fab_box = fab.box();
-    
+
+
+    std::cout<<"FAB box "<<fab_box<<"\n";
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:153]: cout to squash

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp [hunk 4]
```diff
@@ -68,49 +167,50 @@ void FabToBlueprintTopology(const Geometry& geom,
...
     float64 z_max = dims > 2 ? level_z_min + level_dz * k_max : 0.0;
 
-    // create rectilinear coordset 
+    // create rectilinear coordset
     // (which also holds all implicit details needed for the topology)
     res["coordsets/coords/type"] = "rectilinear";
     res["coordsets/coords/values/x"] = DataType::float64(nx+1);
     res["coordsets/coords/values/y"] = DataType::float64(ny+1);
-    
+    std::cout<<"NX "<<nx<<" NY "<<ny<<"\n";
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:182]: and a few more

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp [hunk 5]
```diff
@@ -295,15 +410,23 @@ MultiLevelToBlueprint (int n_levels,
...
             patch["state/cycle"] = level_steps[0];
             patch["state/time"] = time_value;
 
             const FArrayBox &fab = mf[mfi];
+
             // create coordset and topo
             FabToBlueprintTopology(geom,fab,patch);
+            // add the nesting relationship
+            conduit::Node &nestset = patch["nestsets/nest"];
+            std::cout<<"*********** domain "<<domain_id<<" level "<<i<<"\n";
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:426]: and another one

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp (hunk 2 above)
Comment (mclarsen) [reply Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:17]: This was here for debugging. This is essentially the function we will have inside of ascent to mark those zones as ghosts.

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.H (hunk 1 above)
Comment (mclarsen) [reply Src/Extern/Conduit/AMReX_Conduit_Blueprint.H:88]: This can exist in the cpp file.

File: Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp [hunk 6]
```diff
@@ -295,15 +417,22 @@ MultiLevelToBlueprint (int n_levels,
...
 
             const FArrayBox &fab = mf[mfi];
+
             // create coordset and topo
             FabToBlueprintTopology(geom,fab,patch);
+            // add the nesting relationship
+            conduit::Node &nestset = patch["nestsets/nest"];
+            patch["topologies"].print();
+            Nestsets(i, n_levels, fab, box_arrays, ref_ratio, box_offsets, nestset);
+            paint_2d_nestsets(patch,"topo");
```
Comment (cyrush) [inline Src/Extern/Conduit/AMReX_Conduit_Blueprint.cpp:435]: still need to pull out this guy

**Pull request merged**